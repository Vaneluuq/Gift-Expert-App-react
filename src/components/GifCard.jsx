import PropTypes from "prop-types"

const GifCard = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} ></img>
      <p>{title}</p>
    </div>
  )
}

export default GifCard

GifCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
