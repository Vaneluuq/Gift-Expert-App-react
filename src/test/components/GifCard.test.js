import { render, screen } from "@testing-library/react"
import GifCard from "../../components/GifCard"

describe('Pruebas de GifCard componente', () => { 

    const title = "Este es el titulo"
    const url = "https://giphy.com/embed/uBRwFMpdukW1hf8yqS"

    test('debe hacer match con el snapshot ', () => {

        const { container } = render( <GifCard title={title} url={url} /> )
        expect( container ).toMatchSnapshot()
      
    })

    test('Debe de mostrar la imagen con el URL y el alt indicado', () => {
        render( <GifCard title={title} url={url} />)
        // screen.debug() 

        // por cada elemento
        // expect( screen.getByRole("img").src ).toBe( url )
        // expect( screen.getByRole("img").alt ).toBe( title )

        // desestructurando el elemento a evaluar
        const { src, alt } = screen.getByRole("img")
        expect( src ).toBe( url )
        expect( alt ).toBe( title )

    })

    test('Debe de mostrar el titulo en el componente', () => {

        render( <GifCard title={title} url={url} />)
        expect( screen.getByText( title ) ).toBeTruthy()
      
    })
    
    
})