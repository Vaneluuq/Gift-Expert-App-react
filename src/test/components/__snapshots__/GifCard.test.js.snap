// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifCard componente debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Este es el titulo"
      src="https://giphy.com/embed/uBRwFMpdukW1hf8yqS"
    />
    <p>
      Este es el titulo
    </p>
  </div>
</div>
`;
